clock/clock.go

```diff
@@ -24,3 +26,28 @@ func AlignedTick(period time.Duration) <-chan time.Time {
 	}()
 	return c
 }
+
+// AlignedTickLossless returns an aligned ticker that waits for slow receivers,
+// and backfills later as necessary to publish any pending ticks, at possibly
+// a much more aggressive schedule. (keeps ticking until fully caught up)
+// Note: clock jumps may still result in dropped ticks.
+func AlignedTickLossless(period time.Duration) <-chan time.Time {
+	c := make(chan time.Time)
+	nsec := (time.Now().UnixNano() / int64(period)) * int64(period)
+	next := time.Unix(0, nsec).Add(period)
+	go func() {
+		for {
+			now := time.Now()
+			for now.After(next) {
```

Comment:
Might make sense to add a brief comment here noting that this loop is handling the "catch up" ticks (took me a second to parse)